client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -282,10 +282,9 @@ class SimplePaymentsDialog extends Component {
 
 		const { siteId, dispatch, translate } = this.props;
 
-		wpcom
-			.site( siteId )
-			.post( paymentId )
-			.delete()
+		const productAPI = wpcom.site( siteId ).post( paymentId );
+		productAPI.delete()
```

Comment:
`delete( { force: true } )` should do the same thing more elegantly. This is what the API console has to say:

<img width="808" alt="delete-api-console" src="https://user-images.githubusercontent.com/664258/29071996-3fc9f4e2-7c46-11e7-8e6d-d7765f432178.png">

I don't fully understand why we don't just ignore trashed posts on frontend rendering and whether this is supposed to be a permanent solution or just a temporary hack until the Jetpack and the backend are updated, with their slightly longer release cycles.